### PR TITLE
Allow parameter inputs in a template to be expanded to a multi-line input

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1312444

### CODE REVIEW

**Comment by jwforres** [comment]
multi-line-inputs

**Comment by jwforres** [comment]
@spadgett @rhamilto PTAL

@bparees FYI caving for now and putting this on all params so we can quit punting this issue down the road, especially since it sounds like you are adding the generic support for JSON blobs in parameter inputs.  When we know for sure what type of input a parameter should be then we can remove this switch and just set it to the right type of input.

**Comment by bparees** [comment]
works for me.

File: app/views/_templateopt.html [hunk 1]
```diff
@@ -29,7 +30,30 @@
...
           spellcheck="false"
           ng-attr-aria-describedby="{{parameter.description ? (paramID + '-description') : undefined}}">
-    </span>
+      <a href=""
+         ng-click="expandedParameter = !expandedParameter"
+         class="resize-input action-button"
+         data-toggle="tooltip"
+         data-trigger="hover"
+         dynamic-content="{{expandedParameter ? 'Collapse to a single line input. This may strip any new lines you have entered.' : 'Expand to enter multiple lines of content. This is required if you need to include newline characters.'}}"
+        ><i class="fa" ng-class="{'fa-expand': !expandedParemeter, 'fa-compress': expandedParameter}" /></a>
```
Comment (spadgett) [inline app/views/_templateopt.html]: `sr-only` label?

File: app/styles/_core.less [hunk 2]
```diff
@@ -581,7 +581,24 @@ label.checkbox {
     }
     .form-group {
       width: 100%;
-      max-width: 600px;
+
+      .parameter-wrapper {
```
Comment (spadgett) [inline app/styles/_core.less]: Maybe `parameter-input-wrapper` ?

File: app/views/_templateopt.html [hunk 3]
```diff
@@ -29,7 +30,30 @@
...
+      <textarea
+          ng-if="expandedParameter"
+          ng-attr-id="{{paramID}}"
+          ng-attr-name="{{paramID}}"
+          class="form-control hide-ng-leave"
+          type="text"
```
Comment (spadgett) [inline app/views/_templateopt.html]: Shouldn't need `type="text"` for textareas.

**Review by spadgett — COMMENTED** [review]
Some minor comments, otherwise LGTM

**Comment by rhamilto** [comment]
LGTM

**Comment by jwforres** [comment]
@spadgett updated

**Review by spadgett — APPROVED** [review]

**Comment by spadgett** [comment]
[merge]

**Comment by webknjaz** [comment]
FTR, this problem got reintroduced in OpenShift 4: https://github.com/openshift/console/issues/13317.

**Pull request merged**